starlite/handlers/http.py

```diff
@@ -757,6 +791,158 @@ def __init__(
+            include_in_schema: A boolean flag dictating whether  the route handler should be documented in the OpenAPI schema.
+            operation_id: An identifier used for the route's schema operationId. Defaults to the __name__ of the wrapped function.
+            raises:  A list of exception classes extending from starlite.HttpException that is used for the OpenAPI documentation. This list should describe all exceptions raised within the route handler's function/method. The Starlite ValidationException will be added automatically for the schema if any validation is involved.
+            response_description: Text used for the route's response schema description section.
+            security: A list of dictionaries that contain information about which security scheme can be used on the endpoint.
+            summary: Text used for the route's schema summary section.
+            tags: A list of string tags that will be appended to the OpenAPI schema.
+            **kwargs: Any additional kwarg - will be set in the opt dictionary.
+        """
+        if "http_method" in kwargs:
+            raise ImproperlyConfiguredException(MSG_SEMANTIC_ROUTE_HANDLER_WITH_HTTP)
+
+        super().__init__(
+            after_request=after_request,
+            after_response=after_response,
+            background=background,
+            before_request=before_request,
+            cache=cache,
+            cache_control=cache_control,
+            cache_key_builder=cache_key_builder,
+            content_encoding=content_encoding,
+            content_media_type=content_media_type,
+            dependencies=dependencies,
+            deprecated=deprecated,
+            description=description,
+            etag=etag,
+            exception_handlers=exception_handlers,
+            guards=guards,
+            http_method=HttpMethod.HEAD,
+            include_in_schema=include_in_schema,
+            media_type=media_type,
+            middleware=middleware,
+            name=name,
+            operation_id=operation_id,
+            opt=opt,
+            path=path,
+            raises=raises,
+            response_class=response_class,
+            response_cookies=response_cookies,
+            response_description=response_description,
+            response_headers=response_headers,
+            responses=responses,
+            security=security,
+            status_code=status_code,
+            summary=summary,
+            sync_to_thread=sync_to_thread,
+            tags=tags,
+            **kwargs,
+        )
+
+    def _validate_handler_function(self) -> None:
+        """Validates the route handler function once it is set by inspecting
+        its return annotations."""
+        super()._validate_handler_function()
+
+        # we allow here File and FileResponse because these have special setting for head responses
+        if self.signature.return_annotation not in {None, "None", File, "File", FileResponse, "FileResponse"}:
```

Comment:
What if it is a subclass of `File` or `FileResponse`?

Reply:
fixed

Reply:
Unless it's a forward ref of a subclass :wink: